velox/dwio/parquet/tests/ParquetTpchTest.cpp

```diff
@@ -0,0 +1,306 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/dwio/dwrf/test/utils/DataFiles.h"
+#include "velox/dwio/parquet/reader/ParquetReader.h"
+#include "velox/exec/tests/utils/HiveConnectorTestBase.h"
+#include "velox/exec/tests/utils/PlanBuilder.h"
+#include "velox/exec/tests/utils/TempDirectoryPath.h"
+#include "velox/functions/prestosql/registration/RegistrationFunctions.h"
+#include "velox/type/tests/FilterBuilder.h"
+#include "velox/type/tests/SubfieldFiltersBuilder.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::exec;
+using namespace facebook::velox::exec::test;
+
+class ParquetTpchTest : public HiveConnectorTestBase {
+ protected:
+  // Setup a duckDB instance for the entire suite and load TPCH data with scale
+  // factor 0.01.
+  static void SetUpTestCase() {
+    if (duckDb_ == nullptr) {
+      duckDb_ = std::make_shared<DuckDbQueryRunner>();
+      constexpr double tpchsf = 0.01;
+      duckDb_->initTPCH(tpchsf);
+    }
+    functions::prestosql::registerAllFunctions();
+  }
+
+  void SetUp() override {
+    HiveConnectorTestBase::SetUp();
+    parquet::registerParquetReaderFactory();
+    tempDirectory_ = tempDirectory_ = exec::test::TempDirectoryPath::create();
+  }
+
+  void TearDown() override {
+    parquet::unregisterParquetReaderFactory();
+    HiveConnectorTestBase::TearDown();
+  }
+
+  int64_t date(const std::string_view string_date) const {
+    Date date;
+    parseTo(string_date, date);
+    return date.days();
+  }
+
+  // For a given file at filePath, add splits equal to numSplits.
+  std::vector<std::shared_ptr<connector::hive::HiveConnectorSplit>> makeSplits(
+      const std::string& filePath,
+      int64_t numSplits) const {
+    LocalReadFile lfs(filePath);
+    const int fileSize = lfs.size();
+    const int splitSize = fileSize / numSplits;
+    std::vector<std::shared_ptr<connector::hive::HiveConnectorSplit>> splits;
+
+    // Add all the splits except the last one.
```

Comment:
I think you can just add all splits in a loop. No need to treat last split in any special way.